lib/jekyll-ical-tag.rb

```diff
@@ -50,18 +50,41 @@ def render(context)
         length = events.length
 
         events.each_with_index do |event, index|
-          context["event"] = {
-            "index" => index,
-            "uid" => event.uid.presence,
-            "summary" => as_utf8(event.summary).presence,
-            "description" => as_utf8(event.description).presence,
-            "simple_html_description" => as_utf8(event.simple_html_description).presence,
-            "location" => as_utf8(event.location).presence,
-            "url" => as_utf8(event.url&.to_s.presence || event.description_urls.first).presence,
-            "start_time" => event.dtstart&.to_time.presence,
-            "end_time" => event.dtend&.to_time.presence,
-            "attendees" => event.attendees,
-          }
+          # Init
+          context["event"] = {}
+
+          # Jekyll helper variables
+          context["event"]["index"] = index
+
+          # RFC 5545 conformant and custom properties.
+          context["event"].merge!(event.all_properties)
+
+          # Supported but non-standard attributes.
+          context["event"]["attendees"] = event.attendees
+          context["event"]["simple_html_description"] = event.simple_html_description
+
+          # Overridden values
+          context["event"]["url"] ||= event.description_urls.first
+
+          # Deprecated attribute names.
+          context["event"]["end_time"] = context["event"]["dtend"]
+          context["event"]["start_time"] = context["event"]["dtstart"]
+
+          # Ensure all event values are utf8 encoded strings
+          # Ensure times (from dates)
+          # Ensure present
+          context["event"].transform_values do |value|
+            v = case value
+              when String
+                value.force_encoding("UTF-8")
+              when Date
+                value.to_time
+              else
+                value
+              end
+
+            v.presence
+          end
```

Comment:
Forcing encoding to utf8

Reply:
Line 78 in this file should be:

```suggestion
when String, Icalendar::Values::Text
```

See [this comment](https://github.com/Rakefire/jekyll-ical-tag/pull/16#issuecomment-611053355).